build_api_docs.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2020 The qsim Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tool to generate external api_docs for qsim shameless copy from TFQ."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+
+from absl import app
+from absl import flags
+from tensorflow_docs.api_generator import doc_controls
+from tensorflow_docs.api_generator import generate_lib
+from tensorflow_docs.api_generator import public_api
+
+import qsimcirq as qs
+
+flags.DEFINE_string("output_dir", "/tmp/qsim_api", "Where to output the docs")
+
+flags.DEFINE_string("code_url_prefix",
+                    ("https://github.com/quantumlib/qsim/tree/master/"
+                     "qsimcirq"), "The url prefix for links to code.")
+
+flags.DEFINE_bool("search_hints", True,
+                  "Include metadata search hints in the generated files")
+
+flags.DEFINE_string("site_path", "/quark/qsim/api_docs/python",
+                    "Path prefix in the _toc.yaml")
+
+FLAGS = flags.FLAGS
+
+
+def main(unused_argv):
+
+    doc_generator = generate_lib.DocGenerator(
+        root_title="qsim",
+        py_modules=[("qsimcirq", qs)],
+        base_dir=os.path.dirname(qs.__file__),
+        code_url_prefix=FLAGS.code_url_prefix,
+        search_hints=FLAGS.search_hints,
+        site_path=FLAGS.site_path,
+        callbacks=[public_api.local_definitions_filter],
+        private_map={"""Block things from showing up here."""})
```

Comment:
The problem is with this line. This `private_map` argument expects a dict. This is a set. What template did you get this from? we should make this clearer.

If you're not  using this argument, you don't need to pass it. You can just delete this line.

```suggestion
        )
```

Reply:
Switched over to my work machine. Got the same error, with and without the private_map argument. Can anyone else confirm this ?

Reply:
I just tried this locally by cloning from your fork and `pip install  .`, 

As output I got:

```
$ python docs/build_api_docs.py
$ tree /tmp/qsim_api
/tmp/qsim_api
├── index.md
├── qsimcirq
│   ├── _api_cache.json
│   ├── qsim
│   │   ├── add_gate.md
│   │   ├── add_matrix1.md
│   │   ├── add_matrix2.md
│   │   ├── Circuit.md
│   │   ├── GateKind.md
│   │   ├── qsimh_simulate.md
│   │   ├── qsim_sample.md
│   │   ├── qsim_simulate_fullstate.md
│   │   └── qsim_simulate.md
│   ├── qsim_circuit
│   │   └── QSimCircuit.md
│   ├── qsim_circuit.md
│   ├── qsimh_simulator
│   │   └── QSimhSimulator.md
│   ├── qsimh_simulator.md
│   ├── qsim.md
│   ├── qsim_simulator
│   │   ├── QSimSimulator.md
│   │   ├── QSimSimulatorState.md
│   │   └── QSimSimulatorTrialResult.md
│   └── qsim_simulator.md
├── qsimcirq.md
├── _redirects.yaml
└── _toc.yaml
```

Reply:
Interesting.... @95-martin-orion what do you want to do here ?

Reply:
After the changes from @MarkDaoust, I now understand the process to be:
```
# from the top-level qsim directory
virtualenv ~/venv
source ~/venv/bin/activate
pip3 install .
python3 -m pip install git+https://github.com/tensorflow/docs
python3 docs/build_docs.py
tree /tmp/qsim_api
```
I'll try to reproduce this. Seeing different results between users is worrying, but in the end we really only need to worry about whether it works on the devsite.

Reply:
Using the process above, I was able to produce the intended output.